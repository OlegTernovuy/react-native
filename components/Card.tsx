import { FlatList, Text } from 'react-native';

import { CardItem, VerticalList } from '../styled/Card.style';
import { DATA } from '../data/index';

const Card = () => {
    return (
        <VerticalList showsVerticalScrollIndicator={false}>
            <FlatList
                data={DATA}
                renderItem={({ item }) => (
                    <CardItem key={item.id}>
                        <Text>{item.name}</Text>
                    </CardItem>
                )}
                keyExtractor={(item) => item.id}
                contentContainerStyle={{ columnGap: 16 }}
                horizontal
            />
        </VerticalList>
    );
};

export default Card;
