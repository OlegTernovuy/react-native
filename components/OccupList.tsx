import { FlatList, Text } from 'react-native';
import { router } from 'expo-router';
import { useState } from 'react';

import { OccupButton, OccupWrapper } from '../styled/Occup.style';
import { OCCUPATION_TYPE } from '../constants';

const OccupList = () => {
    const [activeOccup, setActiveOccup] = useState('');

    return (
        <OccupWrapper>
            <FlatList
                data={OCCUPATION_TYPE}
                renderItem={({ item }) => (
                    <OccupButton
                        onPress={() => {
                            setActiveOccup(item);
                            router.push(`/search/${item}`);
                        }}
                        item={item}
                        activeOccup={activeOccup}
                    >
                        <Text>{item}</Text>
                    </OccupButton>
                )}
                contentContainerStyle={{ columnGap: 8 }}
                horizontal
            />
        </OccupWrapper>
    );
};

export default OccupList;
