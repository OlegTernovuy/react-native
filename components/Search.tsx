import { Feather } from '@expo/vector-icons';
import { router } from 'expo-router';
import { useState } from 'react';

import {
    SearchButton,
    SearchContainer,
    SearchInput,
    SearchWrapper,
} from '../styled/index.style';

const Search = () => {
    const [searchTerm, setSearchTerm] = useState('');
    return (
        <SearchContainer>
            <SearchWrapper>
                <SearchInput
                    value={searchTerm}
                    onChangeText={(text) => setSearchTerm(text)}
                    placeholder="What are you looking for"
                />
            </SearchWrapper>
            <SearchButton
                onPress={() => {
                    router.push(`/search/${searchTerm}`);
                    setSearchTerm('');
                }}
                testID="search-btn"
            >
                <Feather name="search" size={24} color="black" />
            </SearchButton>
        </SearchContainer>
    );
};

export default Search;
