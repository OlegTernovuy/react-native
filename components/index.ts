import Card from './Card';
import OccupList from './OccupList';
import Search from './Search';

export { Card, OccupList, Search };
