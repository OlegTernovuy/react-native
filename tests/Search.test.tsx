import { fireEvent, render } from '@testing-library/react-native';
import { router } from 'expo-router';

import { Search } from '../components';

jest.mock('expo-router', () => ({
    router: {
        push: jest.fn(),
    },
}));

describe('Search component', () => {
    it('should render successfully', () => {
        const { getByPlaceholderText } = render(<Search />);
        const input = getByPlaceholderText('What are you looking for');
        expect(input).toBeTruthy();
    });

    it('should matches snapshot', () => {
        const { toJSON } = render(<Search />);
        expect(toJSON()).toMatchSnapshot();
    });

    it('should navigate to search page', () => {
        const { getByPlaceholderText, getByTestId } = render(<Search />);
        const searchTerm = 'Von';
        const searchInput = getByPlaceholderText('What are you looking for');

        fireEvent.changeText(searchInput, searchTerm);
        fireEvent.press(getByTestId('search-btn'));

        expect(router.push).toHaveBeenCalledWith(`/search/${searchTerm}`);
    });
});
