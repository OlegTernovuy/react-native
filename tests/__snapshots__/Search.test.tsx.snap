// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search component should matches snapshot 1`] = `
<View
  style={
    {
      "flexDirection": "row",
      "height": 50,
      "marginTop": 8,
    }
  }
>
  <View
    style={
      {
        "backgroundColor": "#f1f1f1cc",
        "borderBottomLeftRadius": 16,
        "borderBottomRightRadius": 16,
        "borderTopLeftRadius": 16,
        "borderTopRightRadius": 16,
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "height": "100%",
        "marginRight": 4,
      }
    }
  >
    <TextInput
      onChangeText={[Function]}
      placeholder="What are you looking for"
      style={
        {
          "height": "100%",
          "paddingBottom": 4,
          "paddingLeft": 4,
          "paddingRight": 4,
          "paddingTop": 4,
          "width": "100%",
        }
      }
      value=""
    />
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#b1d0ffcc",
        "borderBottomLeftRadius": 16,
        "borderBottomRightRadius": 16,
        "borderTopLeftRadius": 16,
        "borderTopRightRadius": 16,
        "height": "100%",
        "justifyContent": "center",
        "opacity": 1,
        "width": 50,
      }
    }
    testID="search-btn"
  >
    <Text
      allowFontScaling={false}
      selectable={false}
      style={
        [
          {
            "color": "black",
            "fontSize": 24,
          },
          undefined,
          {
            "fontFamily": "feather",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          {},
        ]
      }
    >
      
    </Text>
  </View>
</View>
`;
